File: 75-SortColors/75-SortColors.py
# Last updated: 5/29/2026, 11:59:37 AM
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        l,m,h=0,0,len(nums)-1
        while(m<=h):
            if nums[m]==0:
                nums[l],nums[m]=nums[m],nums[l]
                m+=1
                l+=1
            elif nums[m]==1:
                m+=1
            else:
                nums[h],nums[m]=nums[m],nums[h]
                h-=1
        return nums
        

File: 75-SortColors/75-SortColors.java
// Last updated: 5/29/2026, 11:59:44 AM
class Solution {
    public void sortColors(int[] nums) {
        int low = 0, mid = 0, high = nums.length - 1;
        
        while (mid <= high) {
            if (nums[mid] == 0) {
                // swap nums[low] and nums[mid]
                int tmp = nums[low];
                nums[low] = nums[mid];
                nums[mid] = tmp;
                low++;
                mid++;
            } else if (nums[mid] == 1) {
                mid++;
            } else { // nums[mid] == 2
                // swap nums[mid] and nums[high]
                int tmp = nums[mid];
                nums[mid] = nums[high];
                nums[high] = tmp;
                high--;
            }
        }
    }
}
